esitoolsupport/esibase.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2009-2015 Eucalyptus Systems, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses/.
+#
+# Please contact Eucalyptus Systems, Inc., 6755 Hollister Ave., Goleta
+# CA 93117, USA or visit http://www.eucalyptus.com/licenses/ if you need
+# additional information or have any questions.
+
+#
+# Order matters here. We want to make sure we initialize logging before anything
+# else happens. We need to initialize the logger that boto will be using.
+#
+
+import sys
+import subprocess
+import os
+from urlparse import urlparse
+
+
+class EsiBase(object):
+
+    def __init__(self, region='localhost'):
+        self.vars = {}
+        self.region = region
+        self._load_vars()
+        self.check_environment()
+
+    def list_system_accounts(self):
+        accounts = {}
+        process = subprocess.Popen(['/usr/bin/euare-accountlist',
+                                    '-U', self.vars["AWS_IAM_URL"],
+                                    '-I', self.vars["AWS_ACCESS_KEY_ID"],
+                                    '-S', self.vars["AWS_SECRET_ACCESS_KEY"]],
+                                   stdout=subprocess.PIPE)
+        t = process.communicate()
```

Comment:
See my previous commentary about shell injection and passing credentials with environment variables